src/lib/logical_query_plan/lqp_utils.hpp

```diff
@@ -5,24 +5,67 @@
 #include <queue>
 #include <set>
 #include <unordered_set>
+#include <utility>
+#include <vector>
 
 #include "logical_query_plan/abstract_lqp_node.hpp"
-#include "logical_query_plan/aggregate_node.hpp"
-#include "logical_query_plan/alias_node.hpp"
-#include "logical_query_plan/join_node.hpp"
-#include "logical_query_plan/limit_node.hpp"
-#include "logical_query_plan/predicate_node.hpp"
-#include "logical_query_plan/projection_node.hpp"
-#include "logical_query_plan/sort_node.hpp"
-#include "logical_query_plan/update_node.hpp"
 
 namespace opossum {
 
 class AbstractExpression;
+class AbstractLQPNode;
+class LQPSubqueryExpression;
+
 enum class LQPInputSide;
 
 using LQPMismatch = std::pair<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<const AbstractLQPNode>>;
 
+/**
+ * Data structure that maps LQPs to (multiple) subquery expressions that reference them.
+ *
+ * Purpose:
+ *  Mainly used by optimizer rules to optimize subquery LQPs more efficiently. In concrete, it helps
+ *  to optimize subquery LQPs ONLY ONCE, although being referenced by a list of subquery expressions.
+ *
+ * Why weak pointers for subquery expressions?
+ *  Referenced LQPs and subquery expressions might be objects to change after creating this data structure. Depending
```

Comment:
What do you mean with "objects to change"? Any chance you meant "subject (no plural) to change"?

Reply:
I meant "subject to change" but did not formulate it properly ;)